code/modules/food/beverages/bottler.dm

```diff
@@ -0,0 +1,450 @@
+
+/obj/machinery/bottler
+	name = "bottler unit"
+	desc = "A machine that combines ingredients and bottles the resulting beverages."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "bottler_off"
+	density = 1
+	anchored = 1
+	var/obj/item/obj_1 = null
+	var/obj/item/obj_2 = null
+	var/obj/item/obj_3 = null
+	var/list/datum/bottler_recipe/available_recipes
+	var/list/acceptable_items
+	var/glass_bottles = 10
+	var/plastic_bottles = 20
+	var/cans = 25
+	var/bottling = 0
+
+/obj/machinery/bottler/New()
+	if(!available_recipes)
+		available_recipes = new
+		acceptable_items = new
+		//These are going to be acceptable even if they aren't in a recipe
+		acceptable_items |= /obj/item/weapon/reagent_containers/food/snacks
+		acceptable_items |= /obj/item/weapon/reagent_containers/food/drinks/cans
+		//the rest is based on what is used in recipes so we don't have people destroying the nuke disc
+		for (var/type in subtypesof(/datum/bottler_recipe))
+			var/datum/bottler_recipe/recipe = new type
+			if(recipe.result) // Ignore recipe subtypes that lack a result
+				available_recipes += recipe
+				acceptable_items |= recipe.obj_1_item
+				acceptable_items |= recipe.obj_2_item
+				acceptable_items |= recipe.obj_3_item
+			else
+				qdel(recipe)
+
+/obj/machinery/bottler/attackby(obj/item/O as obj, mob/user as mob, params)
+	if(istype(O, /obj/item/weapon/wrench))		//This being before the NODROP check allows borgs to (un)wrench bottlers in case they need move them to fix stuff
```

Comment:
Wasn't aware we had a helper for that, I just copied that check and block of code from existing machinery. Will fix